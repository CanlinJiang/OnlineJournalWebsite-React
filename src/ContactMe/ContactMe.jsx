import React from 'react'
import Frame from "../UtilitiesComponents/MainFrame/Frame"
import styles from "./ContactMe.module.css"
import {useState} from "react"


export default function ContactMe() {
    
    const [submittedForm, setSubmittedForm] = useState();

    const onSubmit = (e) => {
      e.preventDefault();
      const data = new FormData(e.target);
      setSubmittedForm(data);
      console.log("DATA");
    };

    const BodyComponent = () => (
        submittedForm? (
            <div className={styles.form}>
                <div className={styles.left}>
                    <p>{submittedForm.get("firstName")}</p>
                    <p>{submittedForm.get("lastName")}</p>
                    <p>{submittedForm.get("email")}</p>
                </div>
                <p className={styles.right}>{submittedForm.get("message")}</p>
            </div>
        ) : (
        <form id="messageForm" className={styles.form} onSubmit={onSubmit}>
            <div className={styles.left}>
                <input type="text" placeholder="First Name:" name="firstName" ></input>
                <input type="text" placeholder="Last Name:" name="lastName"></input>
                <input type="text" placeholder="Email:" name="email"></input>
                <input type="submit" className={styles.button}></input>
             </div>
        <textarea className={styles.right} form="messageForm" name="message"> Enter message here...</textarea>
        </form>
        )

    
    )

    return (
        <div>
            <Frame BodyComponent={BodyComponent}></Frame>
        </div>
    )
}
