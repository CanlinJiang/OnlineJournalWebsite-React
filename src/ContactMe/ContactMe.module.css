.left {
    width: 100%;
    height: 300px;
    display: flex;
    flex-flow: column nowrap;
    justify-content: flex-start;
    
}

.left > *{
    margin-bottom: 10px;
    border: 2px solid black;
    border-radius: 5px;
    padding: 10px 20px;
}

.left > *:last-child{
    margin-bottom: 0px;
}

.right {
    padding: 20px;
    width: 100%;
    border: 2px solid black;
    border-radius: 5px;
}

.form {
    margin: 150px 200px 200px 200px;
    display: grid;
    grid-template-columns: 1fr 5fr;
    grid-gap: 20px;
}

.button:hover {
    background-color: black;
    color: white;
}

@media (max-width: 767px) {
    .form {
        display: block;
        margin: 30px;
    }
    .left {
        display: block;
        width: 100%;
        height: inherit;
        margin-bottom: 20px;
    }
    .left input{
        width: 100%;
    }
    .right{
        height: 200px;
    }

}

@media (min-width: 768px) and (max-width: 1023px){
    .form{
        margin: 100px;
        margin-bottom: 300px;
    }
    .right{
        height: 200px;
    }
}
