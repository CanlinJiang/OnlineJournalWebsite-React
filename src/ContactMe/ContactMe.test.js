import ContactMe from "./ContactMe"
import React from "react";

describe("ContactMe tests", () => {
    it("renders correctly", () => {
        
      const { container } = render(<ContactMe />);
      expect(container).toMatchSnapshot();
    });
  
  });