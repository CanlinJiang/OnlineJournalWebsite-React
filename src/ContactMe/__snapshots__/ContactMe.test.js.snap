// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMe tests renders correctly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <nav
          class="navigation"
        >
          <ul
            class="leftBar"
          >
            <li>
              <a
                href="/minelist"
              >
                Mine's
              </a>
            </li>
            <li>
              <a
                href="/moodchoosing"
              >
                Moods
              </a>
            </li>
            <li>
              <a
                href="/randomwalk"
              >
                Random Walk
              </a>
            </li>
          </ul>
          <ul
            class="rightBar"
          >
            <li
              id="home"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              id="addNewArticle"
            >
              <a
                href="/addnewarticle"
              >
                Add New Article
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="body"
      >
        <form
          class="form"
          id="messageForm"
        >
          <div
            class="left"
          >
            <input
              name="firstName"
              placeholder="First Name:"
              type="text"
            />
            <input
              name="lastName"
              placeholder="Last Name:"
              type="text"
            />
            <input
              name="email"
              placeholder="Email:"
              type="text"
            />
            <input
              class="button"
              type="submit"
            />
          </div>
          <textarea
            class="right"
            form="messageForm"
            name="message"
          >
             Enter message here...
          </textarea>
        </form>
      </div>
      <div
        class="footer"
      >
        <nav>
          <ul
            class="navigation"
          >
            <li
              id="aboutMe"
            >
              <a
                href="/aboutme"
              >
                About me
              </a>
            </li>
            <li
              id="contactMe"
            >
              <a
                href="/contactme"
              >
                Contact me
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
